modules/icon-theme/hm.nix

```diff
@@ -0,0 +1,121 @@
+{ pkgs, lib, config, ... }:
+
+let
+  cfg = config.stylix.targets.iconTheme;
+in
+{
+  options.stylix.targets.iconTheme = {
+    enable = config.lib.stylix.mkEnableTarget "iconTheme" true;
+    name = lib.mkOption {
+      description = "The icon theme name";
+      type = lib.types.str;
+      default = "Numix-Circle";
+    };
+    package = lib.mkOption {
+      description = "The icon theme package";
+      type = lib.types.package;
+      default = pkgs.numix-icon-theme-circle;
+    };
+    adjustColorScheme = {
+      enable = lib.mkOption {
+        description = "Whether to adjust the icon theme colors.";
+        type = lib.types.bool;
+        default = true;
+      };
+      mode = lib.mkOption {
+        description = ''The mode to use:
+          color = single color
+          color-from-palette = uses a random single color from the palette for each icon
+          palette = uses the entire colorscheme for each icon"
+        '';
+        type = lib.types.enum [ "color" "color-from-palette" "palette" ];
+        default = "palette";
+      };
+      colors = lib.mkOption {
+        description = "The color to use when mode is set to color";
+        type = lib.types.listOf (lib.types.str);
+        default =
+          if cfg.adjustColorScheme.mode == "color" then
+            config.lib.stylix.colors.withHashtag.base09
+          else if cfg.adjustColorScheme.mode == "color-from-palette" then
+            with config.lib.stylix.colors.withHashtag; [
+              base08
+              base09
+              base0A
+              base0B
+              base0C
+              base0D
+              base0E
+              base0F
+            ]
+          else
+            with config.lib.stylix.colors.withHashtag; [
+              base00
+              base01
+              base02
+              base03
+              base04
+              base05
+              base06
+              base07
+              base08
+              base09
+              base0A
+              base0B
+              base0C
+              base0D
+              base0E
+              base0F
+            ];
```

Comment:
Does this need to be an option? It looks more like an implementation detail.

Reply:
I added it in 32da08c497f144d325af7501f2027335db82498b
I just thought it should be customizable, as someone might want to use different colors. Maybe for some color-schemes only some colors work well.